src/cpp/ext/otel/otel_server_call_tracer.cc

```diff
@@ -57,13 +57,11 @@ namespace {
 class OpenTelemetryServerCallTracer : public grpc_core::ServerCallTracer {
  public:
   explicit OpenTelemetryServerCallTracer(const grpc_core::ChannelArgs& args)
-      : start_time_(absl::Now()) {
-    for (const auto& plugin_option :
-         OpenTelemetryPluginState().plugin_options) {
-      active_plugin_options_.push_back(plugin_option != nullptr &&
-                                       plugin_option->IsActiveOnServer(args));
-    }
-  }
+      : start_time_(absl::Now()),
+        active_plugin_options_view_(
+            ActivePluginOptionsView::MakeForServer(args)),
+        injected_labels_from_plugin_options_(
+            OpenTelemetryPluginState().plugin_options.size()) {}
```

Comment:
This could be wasteful if there are a large number of plugin options and only a small subset of them are active.

Note that since the channel args should be the same for all calls on a server, we should be able to determine the number (and set of) active options for the connection and not have to do it per-call.  But I'm guessing we can't actually do that until we implement the rest of the stats plugin infrastructure, so maybe just add a TODO for now?

(Technically, the channel args could actually be different for each individual connection on a given server.  But if we only match against the top-level channel args for the server once when the server starts up and then use the resulting set of active options for all calls on all connections, we can ignore those differences.)

Reply:
Yeah, exactly. TODO added